crates/primitives/starknet/tests/transaction_test.rs

```diff
@@ -0,0 +1,93 @@
+use core::str::FromStr;
+use std::collections::HashMap;
+
+use blockifier::execution::contract_class::ContractClass;
+use blockifier::state::cached_state::CachedState;
+use blockifier::transaction::objects::AccountTransactionContext;
+use frame_support::bounded_vec;
+use hex::FromHex;
+use mp_starknet::execution::{CallEntryPointWrapper, EntryPointTypeWrapper};
+use mp_starknet::state::DictStateReader;
+use mp_starknet::transaction::types::{Transaction, TxType};
+use sp_core::{H256, U256};
+use starknet_api::api_core::{
+    ClassHash, ContractAddress as StarknetContractAddress, ContractAddress, Nonce, PatriciaKey,
+};
+use starknet_api::hash::{StarkFelt, StarkHash};
+use starknet_api::transaction::{Fee, InvokeTransactionV1, TransactionHash, TransactionSignature, TransactionVersion};
+use starknet_api::{patricia_key, stark_felt};
+
+#[test]
+
+fn test_verify_nonce() {
```

Comment:
please move test to `pallets/starknet/tests.rs` and use the mocked state constructed in `fn new_test_ext` in `mock.rs`

Reply:
I couldn't get this. How to retrieve the mock state from the test environment created in `fn new_test_ext` ?

Reply:
I suggest you look at other tests in the `starknet` pallet tests, you can just copy paste one and start from there